src/strategies/predefined.zsh

```diff
@@ -0,0 +1,99 @@
+# zsh-augosuggestion: predefined strategy
+# 1. search in history, returns result if it exists
+# 2. search in predefined files
+#
+# configuration:
+# 
+# ZSH_AUTOSUGGEST_PREDEFINE_NAME  - auto generated predefine file name
+# ZSH_AUTOSUGGEST_PREDEFINE_PATH  - user defined files separated by semicolon
+#
+# "$HOME/.zsh_autosuggest" will be generated at the first time
+#
+# zsh-autosuggestions/predefined.txt is generated from tldr pages and
+# will ship with predefined.zsh .
+#
+
+_zsh_autosuggest_script_path="${0:A:h}"
+
+_zsh_autosuggest_predefined_generate() {
```

Comment:
Seems that this function will lock prompt until it's finished, which is poor user experience. z-asug already has async support, and this task should be started in background.

Reply:
@z0rc this plugin does not have correctly working async func but yes locking is poor experience